configure.ac

```diff
@@ -3526,10 +3526,7 @@ AC_ARG_WITH(soname,
 	        RUBY_SO_NAME='$(RUBY_BASE_NAME)$(MAJOR)$(MINOR)0'
 	    ],
 	    [mingw*], [
-		RUBY_SO_NAME="${rb_cv_msvcrt}"'-$(RUBY_BASE_NAME)$(MAJOR)$(MINOR)0'
-		AS_IF([test x"${target_cpu}" != xi386], [
-		    RUBY_SO_NAME="${target_cpu}-${RUBY_SO_NAME}"
-		])
+	        RUBY_SO_NAME="${target_cpu}-${rb_cv_msvcrt}"'-$(RUBY_BASE_NAME)$(MAJOR)$(MINOR)0'
```

Comment:
Why adding `${target_cpu}-` always, even to x86 platform?

Reply:
without it, building gems will always fail on x86 ucrt..

Reply:
How does it fail?

Reply:
```
building spec/ruby/optional/capi/ext/boolean_spec.so
ld.lld: error: undefined symbol: __initialize_onexit_table
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:64
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(_pre_c_init)

ld.lld: error: undefined symbol: __execute_onexit_table
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:130
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(__CRT_INIT@12)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:130
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(___DllMainCRTStartup)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:130
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(___DllMainCRTStartup)

ld.lld: error: undefined symbol: __amsg_exit
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:94
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(__CRT_INIT@12)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:126
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(__CRT_INIT@12)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:126
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(___DllMainCRTStartup)
>>> referenced 1 more times

ld.lld: error: undefined symbol: __initterm
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:100
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(__CRT_INIT@12)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:104
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(__CRT_INIT@12)

ld.lld: error: undefined symbol: __register_onexit_function
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:207
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(_atexit)

ld.lld: error: undefined symbol: _memcpy
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:294
>>>               libmingw32.a(lib32_libmingw32_a-pseudo-reloc.o):(___write_memory)

ld.lld: error: undefined symbol: ___acrt_iob_func
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:150
>>>               libmingw32.a(lib32_libmingw32_a-pseudo-reloc.o):(___report_error)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:151
>>>               libmingw32.a(lib32_libmingw32_a-pseudo-reloc.o):(___report_error)

ld.lld: error: undefined symbol: _fwrite
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:150
>>>               libmingw32.a(lib32_libmingw32_a-pseudo-reloc.o):(___report_error)

ld.lld: error: undefined symbol: _vfprintf
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:151
>>>               libmingw32.a(lib32_libmingw32_a-pseudo-reloc.o):(___report_error)

ld.lld: error: undefined symbol: _abort
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pseudo-reloc.c:157
>>>               libmingw32.a(lib32_libmingw32_a-pseudo-reloc.o):(___report_error)

ld.lld: error: undefined symbol: _calloc
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/tlsthrd.c:48
>>>               libmingw32.a(lib32_libmingw32_a-tlsthrd.o):(____w64_mingwthr_add_key_dtor)

ld.lld: error: undefined symbol: _free
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/tlsthrd.c:87
>>>               libmingw32.a(lib32_libmingw32_a-tlsthrd.o):(____w64_mingwthr_remove_key_dtor)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/tlsthrd.c:141
>>>               libmingw32.a(lib32_libmingw32_a-tlsthrd.o):(___mingw_TLScallback)

ld.lld: error: undefined symbol: _strlen
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pesect.c:66
>>>               libmingw32.a(lib32_libmingw32_a-pesect.o):(__FindPESectionByName)

ld.lld: error: undefined symbol: _strncmp
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/pesect.c:79
>>>               libmingw32.a(lib32_libmingw32_a-pesect.o):(__FindPESectionByName)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ld.lld: error: undefined symbol: __initialize_onexit_table
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:64
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(_pre_c_init)

ld.lld: error: undefined symbol: __execute_onexit_table
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:130
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(__CRT_INIT@12)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:130
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(___DllMainCRTStartup)
>>> referenced by C:/M/B/src/mingw-w64/mingw-w64-crt/crt/crtdll.c:130
>>>               D:/M/msys64/clang32/lib/dllcrt2.o:(___DllMainCRTStartup)
```